### PR TITLE
Removed runtime flag "do_not_count_mapped_pages_as_free"

### PR DESCRIPTION
**Pull request by DiazAlan**
Commit Message: Removed runtime flag "envoy.reloadable_features.do_not_count_mapped_pages_as_free" and legacy code paths as per issue [#31972](https://github.com/envoyproxy/envoy/issues/31972). It has been over six months since flag was introduced.
Risk Level: Low
Testing: Updated tests to not include legacy code paths.
Release Notes: Add release note to current changelog about removal.

### CODE REVIEW

**Comment by DiazAlan** [comment]
@alyssawilk @KBaichoo PTAL

File: changelogs/current.yaml
Comment (adisuissa) [inline changelogs/current.yaml:31]: Shouldn't this be `envoy.reloadable_features.count_unused_mapped_pages_as_free`?

**Review by adisuissa — COMMENTED** [review]
Thanks for the cleanup!
Minor comment, but otherwise LGTM.

File: changelogs/current.yaml
Comment (DiazAlan) [reply changelogs/current.yaml]: Ah, the unfortunate issue when you copy and paste things from old PRs. Fixed.

**Review by adisuissa — APPROVED** [review]
LGTM, thanks!

**Review by KBaichoo — DISMISSED** [review]

**Review by nezdolik — DISMISSED** [review]

**Review by lizan — APPROVED** [review]
Can you resolve the conflict? Once that's in feel free to merge without my re-approval @adisuissa @KBaichoo @nezdolik

**Review by KBaichoo — APPROVED** [review]

**Pull request merged**
